client/src/cbltest/api/syncgateway.py

```diff
@@ -1273,20 +1107,197 @@ async def get_document_revision_public(
+                body["admin_roles"] = admin_roles
+
+            await self._send_request(
+                "put", f"/{db_name}/_user/{name}", JSONDictionary(body)
+            )
+
+    async def delete_user(self, db_name: str, name: str) -> None:
+        """
+        Deletes a user from a Sync Gateway database
+
+        :param db_name: The name of the Database
+        :param name: The username to delete
+        """
+        with self._tracer.start_as_current_span(
+            "delete_user", attributes={"cbl.user.name": name}
+        ):
+            try:
+                await self._send_request("delete", f"/{db_name}/_user/{name}")
+            except CblSyncGatewayBadResponseError as e:
+                if e.code == 404:
+                    # User doesn't exist, that's fine
+                    pass
+                else:
+                    raise
+
+    async def add_role(self, db_name: str, role: str, collection_access: dict) -> None:
+        """
+        Adds the specified role to a Sync Gateway database with the specified collection access
+
+        :param db_name: The name of the Database to add the user to
+        :param role: The role to add
+        :param collection_access: The collections to which role members will have access.
+            This needs to be formatted in the way Sync Gateway expects it:
+            "<scope1>": {
+                "<collection1>: {"admin_channels" : ["<channel1>", ... ] }
+                .
+                .
+                .
+            }
+            "<scope2>": {
+                ...
+            }
+            .
+            .
+            .
+        """
+        with self._tracer.start_as_current_span(
+            "add_role", attributes={"cbl.role.name": role}
+        ):
+            body = {"collection_access": collection_access}
+
+            await self._send_request(
+                "put", f"/{db_name}/_role/{role}", JSONDictionary(body)
+            )
+
+    async def create_user_client(
+        self,
+        db_name: str,
+        username: str,
+        password: str,
+        channels: list[str],
+    ) -> "SyncGatewayUserClient":
+        """
+        Helper method to create a user with channel access and return a user-specific SG client.
+
+        This is a convenience method for tests that need to verify user-level access control.
+
+        :param db_name: The database name
+        :param username: The username to create
+        :param password: The password for the user
+        :param channels: List of channels the user should have access to
+        :return: A SyncGatewayUserClient instance authenticated as the user (uses public port 4984)
+        """
+        # Clean up user if exists from previous run
+        await self.delete_user(db_name, username)
+        await self.add_user(
+            db_name,
+            username,
+            password=password,
+            collection_access={"_default": {"_default": {"admin_channels": channels}}},
+        )
```

Comment:
I'm not sure this is appropriate to have here.  I think this logic should be separate.